packages/xo-web/src/common/report-bug-button.js

```diff
@@ -53,26 +74,25 @@ const reportOnSupportPanel = async ({
     formData.append('attachments', content, name)
   })
 
-  await Promise.all([
-    timeout.call(xoaUpdater.getLocalManifest(), 5e3).then(
-      manifest =>
-        formData.append(
-          'attachments',
-          createBlobFromString(JSON.stringify(manifest, null, 2)),
-          'manifest.json'
-        ),
-      error => logger.warn('cannot get the local manifest', { error })
-    ),
-    timeout.call(checkXoa(), 5e3).then(
-      xoaCheck =>
-        formData.append(
-          'attachments',
-          createBlobFromString(stripAnsi(xoaCheck)),
-          'xoaCheck.txt'
-        ),
-      error => logger.warn('cannot get the xoa check', { error })
-    ),
-  ])
+  await Promise.all(
+    ADDITIONAL_FILES.map(
+      ({
+        fetch,
+        format = identity,
+        name,
+        timeout: timeout_ = DEFAULT_TIMEOUT,
+      }) =>
+        timeout.call(fetch(), timeout_).then(
```

Comment:
I'd say we could even inline all the "fetches":
- there are not many
- I don't think we need genericity
- it would remove an indirection
- it's more flexible if we want to add a slightly different behaviour (e.g. add a file with the date in its filename)

Let me know what you think, it might just be a personal preference.

Reply:
Thank you for your structured suggestion :+1:

I choose to use a declarative implementation:
- to avoid duplicate the logic `n` times with `n` the number of joined files. Currently, there are `three` files and it's not negligible and we'll probably want soon to add others files. Having a declarative implementation will facilitate adding new files.
- with my implementation we can customize the fetch function and file name.